app/src/main/java/info/nightscout/androidaps/plugins/configBuilder/ProfileFunctionImplementation.kt

```diff
@@ -98,14 +98,17 @@ class ProfileFunctionImplementation @Inject constructor(
         val rounded = time - time % 1000
         // Clear cache after longer use
         synchronized(cache) {
-            if (cache.size() > 30000) {
+            if (cache.keys.size > 30000) {
                 cache.clear()
                 aapsLogger.debug("Profile cache cleared")
             }
-            val cached = cache[rounded]
-            if (cached != null) return cached
+            if (cache.containsKey(rounded)) {
```

Comment:
but old code acted like cache miss on missing data, forcing unneeded db queries. this one clearly stands for `we've searched already and there is nothing`. 
sure this would be better if some kind of sliding expiration were used for cached values... but it'll add complexity